gnu-static.Dockerfile

```diff
@@ -0,0 +1,135 @@
+# syntax=docker/dockerfile:1
```

Comment:
`gnu-static-builder` for consistency with `static-builder` (that we could rename `musl-static-builder` btw)?

Reply:
good idea

Reply:
I think static-builder-musl and static-builder-gnu would be better to keep them grouped together.